### PR TITLE
Use sphinx_rtd theme and separate docs environment

### PR DESCRIPTION
**Pull request by mathbunnyru**
Fix: https://github.com/jupyter/docker-stacks/issues/1597

Alabaster hasn't received any updates in 3 years.
This theme is also not very active, but it was last updated 5 months ago.

I think it looks better and has more modern style.

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -6,5 +6,6 @@ pre-commit
 pytest
 requests
 sphinx
+sphinx_rtd_theme>=1.0.0
```
Comment (mathbunnyru) [inline requirements-dev.txt:9]: In the rtd environment, sphinx_rtd_theme is preinstalled to some old version, which causes troubles.

**Comment by mathbunnyru** [comment]
CI should be a bit faster, also, it's easier to debug pip packages, when only the needed ones are installed.

**Comment by mathbunnyru** [comment]
@trallard please, take a look, I think you will like this :)

**Comment by trallard** [comment]
I was actually going to open an issue to replace the theme for JupyterBook since plenty projects in the Jupyter ecosystem are migrating to this

File: Makefile [hunk 2]
```diff
@@ -132,14 +132,17 @@ dev/%: PORT?=8888
 dev/%: ## run a foreground container for a stack
 	docker run -it --rm -p $(PORT):8888 $(DARGS) $(OWNER)/$(notdir $@)
 
-dev-env: ## install libraries required to build docs and run tests
+install-dev-env: ## install libraries required to build images and run tests
```
Comment (trallard) [inline Makefile:135]: Nice one! I do prefer having the docs and dev requirements separately

**Review by trallard — APPROVED** [review]

**Comment by consideRatio** [comment]
@choldgraf i think you overview the options well, do you have suggestions on a theme to use for a jupyter project in the switch from alabaster?

**Pull request merged**